### PR TITLE
feat: allow version constraints and tag, branch, remote versions for plugin installs [APE-225]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #896
Fixes: APE-225

was there a ticket for allowing version constraints like >= and stuff? i cant find it

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

**Comment by antazoey** [comment]
- [ ] allow cmd line to install git+ style, will need the github action... maybe `ape plugins install vyper@git+https:///` ?

**Comment by antazoey** [comment]
getting these changes in my coverage branch will help me test codecov support!

**Review by fubuloubu — APPROVED** [review]
Great change

File: src/ape_plugins/utils.py [hunk 1]
```diff
@@ -46,12 +54,27 @@ def validate_name(cls, values):
...
+            if f"ape-{name}" not in version:
+                # Just some small validation so you can't put a repo
+                # that isn't this plugin here. NOTE: Forks should still work.
+                raise ValueError("Plugin mismatch with remote git version.")
+
+            version = version
+
+        elif not version:
+            # Only check name for version constraint if not in version.
+            # NOTE: This happens when using the CLI to provider version constraints.
```
Comment (antazoey) [inline src/ape_plugins/utils.py:69]:
```suggestion
            # NOTE: This happens when using the CLI to provide version constraints.
```

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**